pkg/metrics/producers/scheduledcapacity/producer.go

```diff
@@ -25,5 +28,56 @@ type Producer struct {
 
 // Reconcile of the metrics
 func (p *Producer) Reconcile() error {
+	var (
+		loc     *time.Location
+		err     error
+		currVal int32
+	)
+	matchFound := false
+	now := time.Now()
+
+	if p.Spec.Schedule.Timezone != nil {
```

Comment:
We should just default a timezone so that we can assume it's never nil.

Reply:
this lets you collapse some of this logic.

Reply:
Well if they don't specify it, it's nil. Isn't here where we would handle it?

Reply:
IMO, we should have all the default stuff in a single place, right now its spread across multiple files.
We should either add it [here](https://github.com/awslabs/karpenter/blob/main/pkg/apis/autoscaling/v1alpha1/metricsproducer_validation.go#L67) or [here](https://github.com/awslabs/karpenter/blob/main/pkg/apis/autoscaling/v1alpha1/metricsproducer_defaults.go#L20)

Reply:
I agree that we should put all the defaults in one place. I think that would make everything easier. For this PR, I want to keep it here, and we can improve on it in a follow-up.